assets/src/dashboard/app/views/myStories/index.js

```diff
@@ -129,63 +153,109 @@ function MyStories() {
     }
   }, [viewStyle]);
 
-  const filteredStoriesCount = filteredStories.length;
   const listBarLabel = sprintf(
     /* translators: %s: number of stories */
-    _n(
-      '%s total story',
-      '%s total stories',
-      filteredStoriesCount,
-      'web-stories'
-    ),
-    filteredStoriesCount
+    _n('%s total story', '%s total stories', totalStories, 'web-stories'),
+    totalStories
   );
 
   const storiesView = useMemo(() => {
     switch (viewStyle) {
       case VIEW_STYLE.GRID:
         return (
-          <StoryGridView
-            filteredStories={filteredStories}
-            centerActionLabel={
-              <>
-                <PlayArrowIcon />
-                {__('Preview', 'web-stories')}
-              </>
-            }
-            bottomActionLabel={__('Open in editor', 'web-stories')}
-          />
+          orderedStories.length > 0 && (
+            <InfiniteScroller
+              isAllDataLoaded={allPagesLoaded}
+              allDataLoadedMessage={__('No more stories', 'web-stories')}
+              handleGetData={() => {
+                if (!isLoading) {
+                  setCurrentPageToFetch(currentPageToFetch + 1);
+                }
+              }}
+            >
+              <StoryGridView
+                filteredStories={orderedStories}
+                centerActionLabel={
+                  <>
+                    <PlayArrowIcon />
+                    {__('Preview', 'web-stories')}
+                  </>
+                }
+                bottomActionLabel={__('Open in editor', 'web-stories')}
+              />
+            </InfiniteScroller>
+          )
         );
       case VIEW_STYLE.LIST:
         return (
-          <StoryListView
-            filteredStories={filteredStories}
-            storySort={currentStorySort}
-            sortDirection={currentListSortDirection}
-            handleSortChange={setCurrentStorySort}
-            handleSortDirectionChange={setListSortDirection}
-          />
+          orderedStories.length > 0 && (
+            <InfiniteScroller
+              isAllDataLoaded={allPagesLoaded}
+              allDataLoadedMessage={__('No more stories', 'web-stories')}
+              handleGetData={() => {
+                if (!isLoading) {
+                  setCurrentPageToFetch(currentPageToFetch + 1);
+                }
+              }}
```

Comment:
I think it might make sense to make the `infiniteScroller` more responsible for firing an event vs knowing about getting data. think this could be as simple as:
```
          <InfiniteScroller
              canLoadMore={allPagesLoaded}
              isLoading={isLoading}
              allDataLoadedMessage={__('No more stories', 'web-stories')}
              onLoadMore={() => {
                setCurrentPageToFetch(pageIndex =>  pageIndex +  1);
              }}
          >
```